front/src/modules/pathfinding/hooks/usePathfinding.ts

```diff
@@ -209,6 +226,12 @@ export const usePathfinding = (
   }, [vias]);
 
   useEffect(() => {
+    if (
+      (previousOrigin !== origin || previousDestination !== destination) &&
```

Comment:
I don't think this is enough: an intermediary waypoint might've changed too, or rolling stock, or something else…

But, what is the root cause? Why does pathfinding run twice, which state change causes this?

Reply:
intermediary waypoint don’t do aything in this useEffect, and rolling don’t create the bug.

Root cause is pathfinding updating vias when it succeeds. So we need to ignore exactly one call made and resume listening after, which is what `pathStepsCompletedWithPF` does

Reply:
probably gonna change after the refacto https://github.com/OpenRailAssociation/osrd/pull/9924